v0/destinations/af/deleteUsers.js

```diff
@@ -0,0 +1,159 @@
+/* eslint-disable no-await-in-loop */
+/* eslint-disable no-param-reassign */
+const { httpPOST } = require("../../../adapters/network");
+const { generateUUID } = require("../../util");
+const ErrorBuilder = require("../../util/error");
+
+/**
+ * This function is making the ultimate call to delete the user
+ * @param {*} endpoint Endpoint for GDPR request (https://support.appsflyer.com/hc/en-us/articles/360000811585#1-gdpr-request)
+ * @param {*} body default body which is same for all calls
+ * @param {*} identityType type of the identifier those are one of ios_advertising_id, android_advertising_id, appsflyer_id
+ * @param {*} identityValue value of identifier
+ * @returns
+ */
+const deleteUser = async (endpoint, body, identityType, identityValue) => {
+  body.subject_request_id = generateUUID();
+  body.submitted_time = new Date().toISOString();
+  body.subject_identities[0].identity_type = identityType;
+  body.subject_identities[0].identity_value = identityValue;
+  const response = await httpPOST(endpoint, body);
+  return response;
+};
+
+/**
+ * This function will help to delete the users one by one from the userAttributes array.
+ * @param {*} userAttributes Array of objects with userId, emaail and phone
+ * @param {*} config Destination.Config provided in dashboard
+ * @returns
+ */
+const userDeletionHandler = async (userAttributes, config) => {
+  if (!Array.isArray(userAttributes)) {
+    throw new ErrorBuilder()
+      .setMessage("userAttributes is not an array")
+      .setStatus(400)
+      .build();
+  }
+  if (!config?.apiToken || !(config?.appleAppId || config?.androidAppId)) {
+    throw new ErrorBuilder()
+      .setMessage(
+        "API Token and Apple ID or Android App Id are required fields user deletion"
+      )
+      .setStatus(400)
+      .build();
+  }
+  const body = {
+    subject_request_type: "erasure",
+    subject_identities: [{ identity_format: "raw" }]
+  };
+  if (config.statusCallbackUrls) {
+    const statusCallbackUrlsArray = config.statusCallbackUrls.split(",");
+    if (statusCallbackUrlsArray.length > 1) {
```

Comment:
In case of no match, `statusCallbackUrl.match(URLRegex)` returns null --> Are nulls filtered out by default  using `filter` ? Else filteredStatusCallbackUrlsArray length will always be equal to statusCallbackUrlsArray length

Reply:
Add one more test case with 4th entry in statusCallbackUrl array with non-matching regex

Reply:
Nulls are filtered out by default. Added another testcase.